mermithid/processors/misc/TritiumAndEfficiencyBinner.py

```diff
@@ -0,0 +1,152 @@
+Date:3/4/2020
+
+This takes efficiency informations and interpolates between frequency points.
+Then, you dump in tritium data.
+It assigns an efficiency and efficiency uncertainty upper and lower bounds to each event.
+It also bins events and defines an efficiency and efficiency uncertainty upper and
+lower bound by integrating the interpolated efficiency over each bin.
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import scipy
+import sys
+import json
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+from mermithid.misc import Constants
+#from mermithid.misc import ConversionFunctions
+
+logger = morphologging.getLogger(__name__)
+
+
+
+__all__ = []
+__all__.append(__name__)
+
+class TritiumAndEfficiencyBinner(BaseProcessor):
+    '''
+    Processor that takes in tritium data and efficiency data
+    and outputs tritium data (binned and unbinned) with appropriate error
+    bars derived from the efficiency data.
+    Args:
+        energy_or_frequency: type of the tritium input data. Currently this is
+            only set up to use frequency
+        variables: name of the energy/frequency variable (e.g., F or KE)
+        bins: bins to use for the binned output
+        fss_bins: if True, overrides 'bins' and uses the binning for the fss
+            efficiency data
+        efficiency_filepath: path to a json file containing the efficiency and
+            efficiency uncertainties from the fss data analysis
+    Inputs:
+        data: dict containing the unbinned tritium data
+    Output:
+        result: dictionary containing N=the number of events,
+            the bin_efficiencies, the bin_efficiency_errors,
+            the event_efficiencies, and the event_efficiency_errors.
+    '''
+    def InternalConfigure(self, params):
+        '''
+        Configure
+        '''
+
+        # Read other parameters
+        self.namedata = reader.read_param(params, 'variables', "required")
+        self.N = reader.read_param(params, 'N', 'N')
+        self.eff_eqn = reader.read_param(params, 'efficiency', '1')
+        self.bins = reader.read_param(params, 'bins', [])
+        self.asInteger = reader.read_param(params, 'asInteger', False)
+        self.energy_or_frequency = reader.read_param(params, 'energy_or_frequency', 'energy') #Currently only set up to use frequency
+        self.efficiency_filepath = reader.read_param(params, 'efficiency_filepath', 'combined_energy_corrected_eff_at_quad_trap_frequencies.json')
+        self.fss_bins = reader.read_param(params, "fss_bins", False)
+        # If self.fss_bins is True, self.bins is ignored and overwritten
+
+        # initialize the histogram to store the corrected data
+        if self.energy_or_frequency == 'energy':
+            print(sys.getrefcount(self.corrected_data))
+            self.output_bin_variable='KE'
+        elif self.energy_or_frequency == 'frequency':
+            self.output_bin_variable='F'
+
```

Comment:
add a "else" case to return False if it is not energy or frequency: otherwise output_bin_variable will not be defined